rest_framework_bulk/drf3/mixins.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import print_function, unicode_literals
+import traceback
+from rest_framework import status
+from rest_framework.mixins import CreateModelMixin
+from rest_framework.response import Response
+
+
+__all__ = [
+    'BulkCreateModelMixin',
+    'BulkDestroyModelMixin',
+    'BulkUpdateModelMixin',
+]
+
+
+class BulkCreateModelMixin(CreateModelMixin):
+    """
+    Either create a single or many model instances in bulk by using the
+    Serializers ``many=True`` ability from Django REST >= 2.2.5.
+
+    .. note::
+        This mixin uses the same method to create model instances
+        as ``CreateModelMixin`` because both non-bulk and bulk
+        requests will use ``POST`` request method.
+    """
+
+    def create(self, request, *args, **kwargs):
+        bulk = isinstance(request.data, list)
+
+        if not bulk:
+            return super(BulkCreateModelMixin, self).create(request, *args, **kwargs)
+
+        else:
+            serializer = self.get_serializer(data=request.data, many=True)
+            serializer.is_valid(raise_exception=True)
+            self.perform_bulk_create(serializer)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def perform_bulk_create(self, serializer):
+        return self.perform_create(serializer)
+
+
+class BulkUpdateModelMixin(object):
+    """
+    Update model instances in bulk by using the Serializers
+    ``many=True`` ability from Django REST >= 2.2.5.
+    """
+
+    def get_object(self):
+        try:
+            super(BulkUpdateModelMixin, self).get_object()
+        except AssertionError:
+            # probably happened when called get_object() within options()
+            # via self.metadata_class which is not allowed on list viewset
+            # however since we are enabling PUT here, we should handle the
+            # exception if called within options()
+            # We can simply swallow the exception since that method
+            # does not actually do anything with the returned object
+            for file, line, function, code in traceback.extract_stack():
+                if all((file.endswith('rest_framework/views.py'),
+                        function == 'options')):
```

Comment:
> What would be the best way of achieving that?

I'd probably suggest fully overriding `get_object` so that it *actually* has the correct behavior that you need for bulk updates.